src/app/services/board-tiles.service.ts

```diff
@@ -99,35 +103,23 @@ export class BoardTilesService implements ColyseusNotifyable {
 
   constructor(private objectLoader: ObjectLoaderService, private gameState: GameStateService) {}
 
-  initialize(addToScene: (grp: THREE.Group) => void, onProgressCallback: () => void): void {
-    const grp: THREE.Group = this.generateField();
-    addToScene(grp);
-
-    this.tiles = this.gameState.getBoardLayoutAsArray();
-    const tileReadable = [];
-    this.tiles.forEach((tile: Tile) => {
-      if (tile !== undefined) {
-        tileReadable[tile.tileId] = tile.title;
-      }
+  initialize(addToScene: (grp: THREE.Group) => void): Observable<Progress> {
+    return new Observable<Progress>((observer: Observer<Progress>) => {
+      const grp: THREE.Group = this.generateField();
+      addToScene(grp);
+      this.gameState
+        .getBoardLayoutAsArray()
+        .subscribe((tiles: Tile[]) => {
+          this.tiles = tiles;
+          observer.next([1, this.tiles.length + 1]);
+          console.info(
+            'Tiles are:',
+            this.tiles.map((t) => t.title)
+          );
+          this._updateFields(observer);
+        })
+        .unsubscribe();
```

Comment:
what is this trying to achieve

Reply:
nothing - it was meant tin make you think badly of dogs